ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,448 @@
+                    {
+                        Log.Trace($"PsychSignalDataConverter.Convert(): Skipping line {currentLineCount} - Failed to parse date properly");
+                        continue;
+                    }
+                    if (!_mapFileResolver.ResolveMapFile(ticker, timestamp).Any())
+                    {
+                        // Because all tickers are all clustered together, we can detect
+                        // duplicate messages and prevent ourselves from spamming the status log
+                        if (ticker != previousTicker)
+                        {
+                            Log.Trace($"PsychSignalDataDownloader.Convert(): Skipping line {currentLineCount} - Could not resolve map file for ticker {ticker}");
+                        }
+                        previousTicker = ticker;
+                        continue;
+                    }
+
+                    TickerData handle;
+                    if (!_fileHandles.TryGetValue(ticker, out handle))
+                    {
+                        handle = new TickerData(ticker, timestamp.Date, _destinationDirectory);
+                        _fileHandles[ticker] = handle;
+                    }
+
+                    handle.Append(timestamp, csv);
+                    previousTicker = ticker;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Filter raw files to be inclusive within the date range specified and converts those files.
+        /// </summary>
+        /// <param name="startDateUtc">Date to start parsing files from</param>
+        /// <param name="endDateUtc">Date to stop parsing files</param>
+        /// <remarks>Inclusive on the lower bound, and exclusive on the upper bound</remarks>
+        public void ConvertFrom(DateTime startDateUtc, DateTime endDateUtc)
+        {
+            if (_disposedValue)
+            {
+                throw new ObjectDisposedException("PsychSignalDataConverter has already been disposed");
+            }
+
+            // Filter for files by name and bounds, then order by date
+            var files = _rawSourceDirectory.GetFiles("*.csv", SearchOption.TopDirectoryOnly)
+                .Where(
+                    x =>
+                    {
+                        DateTime fileDate;
+                        if (!DateTime.TryParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal, out fileDate))
+                        {
+                            return false;
+                        }
+
+                        return fileDate >= startDateUtc && fileDate < endDateUtc;
+                    }
+                )
+                .OrderBy(x => DateTime.ParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal))
+                .ToList();
+
+            var fileCount = files.Count();
```

Comment:
nit - should use `Count` property instead of `Enumerable.Count()` -- that being said -- the implementation of `Enumerable.Count()` will actually perform a cast to see if it implements `ICollection` and then use the `Count` property of that. If it _doesn't_ implement `ICollection`, the `Enumerable.Count()` method would actually loop over everything to compute the count.